packages/devtools_app/lib/src/logging/logging_controller.dart

```diff
@@ -473,7 +492,19 @@ class LoggingController {
 
   void _log(LogData log) {
     // Insert the new item and clamped the list to kMaxLogItemsLength.
-    data.add(log);
+    if (searchText?.isNotEmpty ?? false) {
```

Comment:
I see this filtering logic in a couple of places. Add a helper method to determine whether a  LogData entry should be in data or rawData.